autoload/lsp/internal/completion/documentation.vim

```diff
@@ -65,43 +68,113 @@ endfunction
 
 function! s:show_floating_window(event, managed_user_data) abort
     let l:completion_item = a:managed_user_data['completion_item']
-    let l:documentation = get(l:completion_item, 'documentation', '')
-    if type(l:documentation) == type({}) && has_key(l:documentation, 'value')
-        let l:documentation = l:documentation['value']
+
+    " Ignore if documentation does not exists.
+    if !has_key(l:completion_item, 'documentation')
+        call s:close_floating_window(v:true)
+        return
     endif
+
+    " Ignore normalized documentation is empty.
+    let l:documentation = s:MarkupContent.normalize(l:completion_item['documentation'])
     if empty(l:documentation)
+        call s:close_floating_window(v:true)
         return
     endif
 
-    if exists('s:floating_win')
-        call s:floating_win.close()
+    " Add detail field if provided.
+    if type(get(l:completion_item, 'detail', v:null)) == type('')
+        if !empty(l:completion_item.detail)
+            let l:detail = s:MarkupContent.normalize({
+            \   'language': &filetype,
+            \   'value': l:completion_item['detail'],
+            \ })
+            let l:documentation = l:detail . "\n\n" . l:documentation
```

Comment:
do we need to check for empty detail?

Reply:
It is not need to check. because `MarkupContent.normalize({ `language': ..., 'value': ... })` will creates fenced code block (so it always has contents.)